packages/swing-store/src/snapStore.js

```diff
@@ -203,70 +167,69 @@ export function makeSnapStore(
    *
    * @param {string} vatID
    * @param {number} snapPos
-   * @param {(filePath: string) => Promise<void>} saveRaw
+   * @param {AsyncIterableIterator<Uint8Array>} snapshot
    * @returns {Promise<SnapshotResult>}
    */
-  async function saveSnapshot(vatID, snapPos, saveRaw) {
+  async function saveSnapshot(vatID, snapPos, snapshot) {
     const cleanup = [];
     return aggregateTryFinally(
       async () => {
-        // TODO: Refactor to use tmpFile rather than tmpName.
-        const tmpSnapPath = await ptmpName({ template: 'save-raw-XXXXXX.xss' });
-        cleanup.push(() => unlink(tmpSnapPath));
-        const { duration: rawSaveSeconds } = await measureSeconds(async () =>
-          saveRaw(tmpSnapPath),
-        );
-        const { size: uncompressedSize } = await stat(tmpSnapPath);
-
-        // Perform operations that read snapshot data in parallel.
-        // We still serialize the stat and opening of tmpSnapPath
-        // and creation of tmpGzPath for readability, but we could
-        // parallelize those as well if the cost is significant.
-        const snapReader = createReadStream(tmpSnapPath);
-        cleanup.push(() => {
-          snapReader.destroy();
-        });
-
-        await fsStreamReady(snapReader);
-
         const hashStream = createHash('sha256');
         const gzip = createGzip();
         let compressedSize = 0;
+        let uncompressedSize = 0;
 
-        const { result: hash, duration: compressSeconds } =
+        const { duration: compressSeconds, result: compressedSnapshot } =
```

Comment:
I know the variable was there in the original, but "compressSeconds".. with the tee that's happening, this isn't really measuring the time it takes to compress the snapshot, right? It's more like the time it takes for the writer to give us the snapshot, plus the time it takes us to compress it?

Eventually we should look at the timers we're using and decide if they're providing us with meaningful data.

Reply:
Correct but since I still expect the compression to be the limiting factor, I didn't feel too bad leave it at `compressSeconds`. A `createAndCompressSeconds` would be more accurate, but that's a mouthful. Open to suggestions.